### PR TITLE
Implemented pod aggregator.

### PR DESCRIPTION
**Pull request by piosz**

### CODE REVIEW

File: processors/pod_aggregator.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+package processors
+
+import (
+	"fmt"
+
+	"k8s.io/heapster/core"
+)
+
+type PodAggregator struct {
+	MetricsToAggregate []string
```
Comment (mwielgus) [inline processors/pod_aggregator.go]: All metrics should be aggregated.

File: processors/pod_aggregator.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	LabelsToPopulate   []core.LabelDescriptor
```
Comment (mwielgus) [inline processors/pod_aggregator.go]: This should rather be a constant.

File: processors/pod_aggregator.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+				for _, metricName := range this.MetricsToAggregate {
+					metricValue, found := metricSet.MetricValues[metricName]
+					if !found {
+						continue
+					}
+					aggregatedValue, found := pod.MetricValues[metricName]
+					if found {
+						if aggregatedValue.ValueType != metricValue.ValueType {
+							return nil, fmt.Errorf("PodAggregator: type not supported in %s", metricName)
```
Comment (mwielgus) [inline processors/pod_aggregator.go]: Inconsistent type.

File: processors/pod_aggregator.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+						}
+
+						if aggregatedValue.ValueType == core.ValueInt64 {
+							aggregatedValue.IntValue += metricValue.IntValue
+						} else if aggregatedValue.ValueType == core.ValueFloat {
+							aggregatedValue.FloatValue += metricValue.FloatValue
+						} else {
+							return nil, fmt.Errorf("PodAggregator: type not supported in %s", metricName)
```
Comment (mwielgus) [inline processors/pod_aggregator.go]: Same here.

File: processors/pod_aggregator.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+						}
+					} else {
+						aggregatedValue = metricValue
+					}
+					pod.MetricValues[metricName] = aggregatedValue
+				}
+			} else {
+				return nil, fmt.Errorf("No namespace and/or pod info in container %s: %v", key, metricSet.Labels)
```
Comment (mwielgus) [inline processors/pod_aggregator.go]: Do not crash.

File: processors/pod_aggregator.go (hunk 1 above)
Comment (piosz) [inline processors/pod_aggregator.go]: Done

File: processors/pod_aggregator.go (hunk 2 above)
Comment (piosz) [inline processors/pod_aggregator.go]: Done

File: processors/pod_aggregator.go (hunk 5 above)
Comment (piosz) [inline processors/pod_aggregator.go]: I don't agree. We should have pod and namespace specified for every pod container. If there is not probably source implementation doesn't work as expected.

File: processors/pod_aggregator.go (hunk 4 above)
Comment (piosz) [inline processors/pod_aggregator.go]: Discussed offline. Leaving as it was.

File: processors/pod_aggregator.go (hunk 3 above)
Comment (piosz) [inline processors/pod_aggregator.go]: Done

**Comment by piosz** [comment]
PTAL

File: processors/pod_aggregator.go (hunk 5 above)
Comment (piosz) [inline processors/pod_aggregator.go]: Discussed offline,.

**Comment by mwielgus** [comment]
LGTM

**Pull request merged**